R/util.R

```diff
@@ -2920,3 +2920,146 @@ separate_japanese_address <- function(df, address, prefecture_col = "prefecture"
   # call tidyr::unnest_wider so that each element becomes dedicated column like prefecture, city, and street.
   df %>% tidyr::unnest_wider(.exploratory_dummy_column_for_japanese_address, names_repair = ~new_names)
 }
+
+# Constructs a new vector string for the levels.
+# It generates a new character vector based on the base.labels
+# and overrides it with new.labels. The return value has the same
+# length as the base.labels.
+#
+#  new.labels <- c("a", "b", "c")
+#  base.labels <- c("1", "2", "3", "4", "5")
+#  _tam_construct_new_labels (base.labels, new.labels)
+#  [1] "a" "b" "c" "4" "5"
+#
+#  new.labels <- c("a", "b", "c", "d", "e", "f")
+#  base.labels <- c("1", "2", "3", "4", "5")
+#  _tam_construct_new_labels (base.labels, new.labels)
+#  [1] "a" "b" "c" "d" "e"
+construct_new_labels <- function(base.labels, new.labels) {
+  if (is.null(new.labels)) {
+    base.labels
+  }
+  # If the length of new.labels and base.labels are the same,
+  # new.labels is ready to use. Just return it.
+  else if (length(new.labels) == length(base.labels)) {
+    coalesce(new.labels, base.labels)
+  }
+
+  # If new.labels is longer than the base.labels, chop it to make them
+  # in the same length and return it.
+  else if (length(new.labels) > length(base.labels)) {
+    length(new.labels) <- length(base.labels)
+    coalesce(new.labels, base.labels)
+  }
+  else {
+    # If new.labels is shorter than the base.labels, pad NAs at the end
+    # of the vector to make them in the same length.
+    v <- c(new.labels, rep(NA, length(base.labels) - length(new.labels)))
+    coalesce(v, base.labels)
+  }
+}
+# Format the values generated by cut function nicely.
+#
+# Example
+# Input: "(0.996,2.33]", "(2.33,3.67]", "(3.67,5]"
+# Output: "0.996 - 2.33", "2.33 - 3.67", "3.67 - 5"
+format_cut_output <- function(x, decimal.digits=2, big.mark=",", small.mark=".", new.labels=NULL, prefix="", suffix="", right=TRUE) {
+  # Exit if the length is 0.
+  if (length(x) == 0 || all(is.na(x))) {
+    return (x)
+  }
+  # Get levels
+  x.levels <- levels(x)
+  # Format the text.
+  new.names <- format_cut_output_line(x.levels, decimal.digits, big.mark, small.mark, prefix, suffix, right)
+
+  if (!is.null(new.labels)) {
+    new.names <- construct_new_labels(new.names, new.labels)
+  }
+
+  # Make it a named vector with the original values.
+  names(new.names) <- x.levels
+  # Recode the original strings with the formatted strings.
+  # Use a named character vector for unquote splicing with !!!.
+  # https://www.rdocumentation.org/packages/dplyr/versions/0.7.8/topics/recode
+  dplyr::recode(x, !!!new.names)
+}
+
+# It formats a given character vector generated by cut function.
+#
+# Example:
+# Input: [1] "(0.996,2.333333333]"       "(2.333333333,3.666666667]" "(3.666666667,5.004]"
+# Output: [1] "1.00 - 2.33" "2.33 - 3.67" "3.67 - 5.00"
+#
+format_cut_output_line <- function(x, decimal.digits=2, big.mark=",", small.mark=".", prefix="", suffix="", right=TRUE) {
```

Comment:
Rename it to something like format_cut_output_levels to express what it is better.

Reply:
Renamed it to format_cut_output_levels